versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksBaseValue.java

```diff
@@ -0,0 +1,215 @@
+
+  private Id id;
+  private long datetime;
+
+  RocksBaseValue() {
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public C id(Id id) {
+    this.id = id;
+    return (C) this;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public C dt(long dt) {
+    this.datetime = dt;
+    return (C) this;
+  }
+
+  Id getId() {
+    return id;
+  }
+
+  /**
+   * Converts a Stream of Ids into a List Entity.
+   * @param idStream the stream of Id to convert.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream) {
+    return Entity.ofList(idStream.map(Id::toEntity).collect(Collectors.toList()));
+  }
+
+  /**
+   * Retrieves an Id at 'position' in a Stream of Ids as an Entity.
+   * @param idStream the stream of Id to convert.
+   * @param position the element in the Stream to retrieve.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream, int position) {
+    return idStream.skip(position).findFirst().orElseThrow(NoSuchElementException::new).toEntity();
+  }
+
+  /**
+   * Evaluates if the stream of Id meets the Condition Function.
+   * @param function the function of the ConditionExpression to evaluate.
+   * @param stream The Ids to evaluate against the function
+   * @return true if the condition is met
+   */
+  boolean evaluateStream(Function function, Stream<Id> stream) {
+    // EQUALS will either compare a specified position or the whole stream as a List.
+    if (function.getOperator().equals(Function.EQUALS)) {
+      final ExpressionPath.PathSegment pathSegment = function.getPath().getRoot().getChild().orElse(null);
+      if (pathSegment == null) {
+        return toEntity(stream).equals(function.getValue());
+      } else { // compare complete list
+        if (pathSegment.isPosition()) { // compare individual element of list
+          final int position = pathSegment.asPosition().getPosition();
+          return toEntity(stream, position).equals(function.getValue());
+        }
+      }
+    } else if (function.getOperator().equals(Function.SIZE)) {
+      return (stream.count() == function.getValue().getNumber());
+    }
+
+    return false;
+  }
+
+  @Override
+  public boolean evaluate(List<Function> functions) {
+    for (Function function: functions) {
+      if (function.getPath().getRoot().isName()) {
+        if (!evaluateFunction(function)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Checks that a Function is met by the implementing class.
+   * @param function the condition to check
+   * @return true if the condition is met
+   */
+  public abstract boolean evaluateFunction(Function function);
+
+  /**
+   * Evaluates the id against a function and ensures that the name segment is well formed,
+   * ie does not contain a child attribute.
+   * @param function the function to evaluate against id.
+   * @return true if the id meets the function condition
+   */
+  boolean idEvaluates(Function function) {
```

Comment:
You'll probably get a comment about the name here, but I don't know a better name.

Reply:
Maybe to swap the name around `evaluatesId` ?

Reply:
Sure, please do so.

Reply:
Done